### DISCUSSION TITLE
Returned Source Documents also include documents not related to the answer

### DISCUSSION THREAD

**Discussion by Valdinia** [FlowiseAI/Flowise#1065] [Q&A]
I'm using a Conversational Retrieval QA Chain.
Its knowledge base (indexed in a Pinecone vector db) is based on just 3 test documents, containing unrelated stories.
The answers given by the chat flow are pretty good for now, and the returned Source Documents do include the correct document and paragraphs the answer is based on, but, for whatever reason they also include the other documents that have nothing to do with the answer. Why is that?! How could I solve this problem?

**Comment by matthias** [comment]
You may have specified to receive at least 4 documents in your chain. In this example it will always return 4 documents. (Note: a LangChain document is not the same as a "document" / pdf -> a source document may be chunked / splited into several (sub)documents.

![Flowise_-_Low-code_LLM_apps_builder](https://github.com/FlowiseAI/Flowise/assets/16365/21cdbc35-526f-46c5-803a-8bf8d6c0418d)

**Comment by HenryHengZJ** [comment]
If you dont have metadata/namespace specified, all vectors will be searched and returned the top K results, thats why you are seeing some unrelated docs
